src/lib/inference_utils.js

```diff
@@ -423,94 +426,70 @@ export function output2BB(output, numImages, minConfidence, nms = false) {
         bestScore : liste des meilleurs scores pour chaque image
             forme : [each img [each box score]]
     */
-	if (nms) {
-		let bestBoxes = [];
-		let bestScores = [];
-		// @ts-ignore
-		let numbb = output.length / 6;
+	/** @type {import('./inference.js').BB[][]}  */
+	let bestBoxes = [];
+	/** @type {number[][]}  */
+	let bestScores = [];
+	/** @type {?Float32Array}  */
+	let suboutput = null;
+
+	console.log(output);
+
+	const outputShape = protocol.inference?.detection.output.shape ?? [
+		'cx',
+		'cy',
+		'w',
+		'h',
+		'score',
+		'_'
+	];
+
+	const indexOffsets = {
+		x: outputShape.findIndex((v) => oneOf(v, ['cx', 'sx'])),
+		y: outputShape.findIndex((v) => oneOf(v, ['cy', 'sy'])),
+		w: outputShape.indexOf('w'),
+		h: outputShape.indexOf('h'),
+		score: outputShape.indexOf('score')
+	};
+	const suboutputSize = outputShape.length;
+	let boundingBoxesCount = output.length / suboutputSize;
+
+	for (let k = 0; k < numImages; k++) {
+		/** @type {import('./inference.js').BB[]}  */
 		let bestPerImageBoxes = [];
-		let _bestScorePerImage = 0;
-		let suboutput = null;
-
-		console.log(output)
-
-		console.log('num images : ', numImages);
-
-		for (let k = 0; k < numImages; k++) {
-			bestPerImageBoxes = [];
-			_bestScorePerImage = 0;
-			/** @type {number[]} */
-			let bbScores = [];
-			// @ts-ignore
-			suboutput = output.slice(k * numbb * 6, (k + 1) * numbb * 6);
-
-			for (let i = 0; i < suboutput.length; i += 6) {
-				let score = suboutput[i + 4];
-				let x = suboutput[i];
-				let y = suboutput[i + 1];
-				let w = suboutput[i + 2];
-				let h = suboutput[i + 3];
-
-				w = Math.abs(x - w);
-				h = Math.abs(y - h);
-
-				if (x == 0 && y == 0 && w == 0 && h == 0) {
-					break;
-				}
-				// if (score > bestScorePerImage) {
-				// 	bestScorePerImage = score;
-				// }
-				if (score > minConfidence) {
-					bestPerImageBoxes.push([x, y, w, h]);
-					bbScores.push(score);
-				}
+		/** @type {number[]} */
+		let bbScores = [];
+
+		suboutput = output.slice(
+			k * boundingBoxesCount * suboutputSize,
+			(k + 1) * boundingBoxesCount * suboutputSize
+		);
+
+		for (let i = 0; i < suboutput.length; i += suboutputSize) {
+			let x = suboutput[i + indexOffsets.x];
+			let y = suboutput[i + indexOffsets.y];
+			let w = suboutput[i + indexOffsets.w];
+			let h = suboutput[i + indexOffsets.h];
+			let score = suboutput[i + indexOffsets.score];
+
+			// TODO understand this, seems like our model does _not_ really output width and height, what does it output then?
```

Comment:
probably ending points for x and y:

$$
\begin{align*}
w &= \text{end}_x - \text{start}_x = |\text{start}_x - \text{end}_x| \\
h &= \text{end}_y - \text{start}_y = |\text{start}_y - \text{end}_y|
\end{align*}
$$

where `w` (resp. `h`) first represents $\text{end}_x$ (resp $\text{end}_y$) before the reassignment, and `x` (resp. `y`) is $\text{start}_x$ (resp. $\text{start}_y$)

Reply:
pour être sûre dcp @CceceLaPlante on est d'accord c bien ça? les w et h = Math.abs() d'apres, enfait les w et h du rezo c'est le point infereieur droit de la BB ?

Reply:
yep ct ça ^^